src/Cookie.php

```diff
@@ -144,37 +177,131 @@ public function httpOnly(bool $httpOnly): self
     public function sameSite(string $sameSite): self
     {
         if (!in_array($sameSite, [self::SAME_SITE_LAX, self::SAME_SITE_STRICT, self::SAME_SITE_NONE], true)) {
-            throw new \InvalidArgumentException('sameSite should be either Lax or Strict');
+            throw new InvalidArgumentException('sameSite should be one of "Lax", "Strict" or "None"');
+        }
+
+        if ($sameSite === self::SAME_SITE_NONE) {
+            // the secure flag is required for cookies that are marked as 'SameSite=None'
+            $this->secure = true;
```

Comment:
Shouldn't you set `secure` to `$new` object?

Reply:
Yep, my mistake will push a fix with further commit. The `Cookie` is immutable, so we don't change any property directly on it.